base/workflow/inst/batch_run.R

```diff
@@ -45,122 +45,119 @@ pecan_path <- get_arg(argv, "--pecandir", getwd())
-    "PEcAn Workflow Complete",
-    raw_output
-  ))
-  continue <- FALSE
-  ##################################################
-  # Did we write a job.sh file?
-  ##################################################
-  out <- file.path(outdir, "out")
-  run <- file.path(outdir, "run")
-  jobsh <- list.files(run, "job\\.sh", recursive = TRUE)
-  ## pft <- file.path(outdir, "pft")
-  if (length(jobsh) > 0) {
-    result_table$has_jobsh[[i]] <- TRUE
-    continue <- TRUE
-  }
 
-  ##################################################
-  # Did the model produce any output?
-  ##################################################
-  if (continue) {
-    continue <- FALSE
-    raw_out <- list.files(out, recursive = TRUE)
-    if (length(raw_out) > 0) {
-      result_table$model_output_raw[[i]] <- TRUE
-      continue <- TRUE
+#----------- Checking the results of the runs
+checks_df<-list.dirs(output_folder, full.names = TRUE, recursive = FALSE) %>% 
+  purrr::map_dfr(function(outdir){
+    
+    result_table <-NULL
+    ##################################################
+    # Did the workflow finish?
+    ##################################################
+    if (file.exists(file.path(outdir, "workflow.Rout"))) {
+      raw_output <- readLines(file.path(outdir, "workflow.Rout"))
+      result_table$workflow_complete <- any(grepl(
+        "PEcAn Workflow Complete",
+        raw_output
+      ))
+    }else{
+      result_table$workflow_complete <- FALSE
     }
-  }
-
-  ##################################################
-  # Did PEcAn post-process the output?
-  ##################################################
-  # Files should have name `YYYY.nc`
-  if (continue) {
-    continue <- FALSE
+    ##################################################
+    # Did we write a job.sh file?
+    ##################################################
+    out <- file.path(outdir, "out")
+    run <- file.path(outdir, "run")
+    jobsh <- list.files(run, "job\\.sh", recursive = TRUE)
+    ## pft <- file.path(outdir, "pft")
+    result_table$has_jobsh <-ifelse(length(jobsh) > 0, TRUE, FALSE)
+    ##################################################
+    # Did the model produce any output?
+    ##################################################
+    raw_out <- list.files(out, recursive = TRUE)
+    result_table$model_output_raw <-ifelse(length(raw_out) > 0, TRUE, FALSE)
+    ##################################################
+    # Did PEcAn post-process the output?
+    ##################################################
+    # Files should have name `YYYY.nc`
     proc_out <- list.files(out, "[[:digit:]]{4}\\.nc", recursive = TRUE)
-    if (length(proc_out) > 0) {
-      result_table$model_output_processed[[i]] <- TRUE
-      continue <- TRUE
-    }
+    result_table$model_output_processed <-ifelse(length(proc_out) > 0, TRUE, FALSE)
+    
+    return(result_table %>% 
+             as.data.frame() %>%
+             mutate(site_id=strsplit(outdir,"_")[[1]][5])
```

Comment:
This seems fragile, since both output directories and site names themselves are _very_ likely to have underscores.

Why not just append to the original result table, like the original version of this code?

Reply:
The fundamental change here for me was to break the main sequential `for` loop into two separate loops, one parallelized for sending out all the jobs and then when they are all done another one for checking the outputs. 
Yeah this piece could be fragile, I'll find a better way.

Reply:
Since you should be able to guess the output directory ahead of time, I would add it as a column to the input table and then `left_join(input_table, result_table, by = "outdir")`. That way, you preserve all of the metadata from the input table.